src/cache/RrMapCache.js

```diff
@@ -0,0 +1,41 @@
+import validateCacheSize from './util/validateCacheSize';
+import isStringOrNumber from './util/isStringOrNumber';
+
+export default class RrMapCache {
+  constructor({cacheSize} = {}) {
+    validateCacheSize(cacheSize);
+    this.clear();
+    this._cacheSize = cacheSize;
+  }
+  set(key, selectorFn) {
+    if (this._cacheLength >= this._cacheSize) {
+      this._randomReplace(key, selectorFn);
+    } else {
+      this._cache.set(key, selectorFn);
+      this._cacheKeys[this._cacheLength] = key;
+      this._cacheLength++;
+    }
+  }
+  get(key) {
+    return this._cache.get(key);
+  }
+  remove(key) {
+    const index = this._cacheKeys.indexOf(key); // O(1)
+    if (index > -1) {
+      delete this._cache.delete(key);
+      this._cacheLength--;
+      this._cacheKeys[index] = this._cacheKeys[this._cacheLength];
+    }
+  }
+  clear() {
+    this._cache = new Map();
+    this._cacheKeys = [];
+    this._cacheLength = 0;
```

Comment:
Yeah that should work fine.

Happy to go either way, but I'll drop some arguments for keeping `_cacheLength`:

1. Keeps the code nearly identical to `RrObjectCache`
2. Explicitly stores the current length, to avoid assumptions and confusion about the difference between `_cache.size` ([presumably equivalent](https://tc39.es/ecma262/#sec-get-map.prototype.size)) and `_cacheKeys.length` (not equivalent)
3. The explicit count is ~1% faster in my benchmarks 🤷‍♂️

Reply:
Unless we run into significant perf issues I'd go for relying on native API's and keep code shorter, when possible.

As for point 1, yes, we'd create a divergence from `RrObjectCache` but we'd make it more similar to the existing Map-based caches. ⚖️ 

Happy to fine tune performances later if we have evidence of such concerns. 👍